src/components/tooltip/index.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { ReactElement } from 'react'
+import Portal from '@reach/portal'
+import { useTooltip, TooltipPopup } from '@reach/tooltip'
+import classNames from 'classnames'
+import { TooltipArrowProps, TooltipProps, PRect } from './types'
+import s from './style.module.css'
+
+/**
+ * OFFSET is the pixel distance to set
+ * between the child element's edge and the tooltip edge.
+ * Note that the tooltip arrow extends above the tooltip edge.
+ */
+const SPACE_FROM_CHILDREN = 12
+
+/**
+ * ARROW_WIDTH is the width of the tooltip arrow.
+ * In an ideal world we might calculate this from the .arrow element size
+ * after its 45 degree rotation has been applied, but it's much
+ * easier to instead hard-code this value.
+ */
+const ARROW_WIDTH = 16
+
+/**
+ * COLLISION_BUFFER is the minimum pixel distance to maintain
+ * between the tooltip edge and the viewport edge.
+ */
+const COLLISION_BUFFER = 8
+
+function Tooltip({
```

Comment:
Much of this `Tooltip` implementation was swiped from `@hashicorp/react-tabs`.

Reply:
Seeing we have some positioning logic included, thought I'd share [floating-ui](https://floating-ui.com/docs/react-dom-interactions#examples) that is a newer library for positioning elements like tooltips, popovers, etc. Built by the same person who built popper.js.

Reply:
😍  oooh wow that looks awesome! Added 🎟️ [[tooltip] Consider FloatingUI for positioning](https://app.asana.com/0/1202097197789424/1202183875131237/f) as a future enhancement 💯